<launch>
    <node pkg="djiros" type="djifox_node" name="djiros" output="screen" launch-prefix="">
        <param name="log_level" type="string" value="info"/>
        <param name="serial_name" type="string" value="/dev/ttyUSB0"/>
        <param name="baud_rate" type="int" value="921600"/>
        <param name="A3_or_M100" type="int" value="0"/>
        <param name="app_id" type="int" value="$(env DJIROS_APPID)"/>
        <param name="app_version" type="int" value="1"/>
        <param name="enc_key" type="string" value="$(env DJIROS_ENCKEY)"/>  
        <param name="only_broadcast" type="bool" value="false"/>
		<param name="use_imu_filter" type="bool" value="true"/>
        <param name="align_with_fmu" type="bool" value="false"/>
        <rosparam>
            # common setting and default individual settings
            verbose_output: false
            cam_cnt: 1
            fps:  20
            gain: 1.0
            use_auto_exposure: false
            fast_mode: false

            # individual setting for camera 0
            camera0:
                serial: "25000049"
                topic: image
                use_auto_exposure: false
                exposure_time_us: 10000
                auto_speed: 0 # 0--1--2, slow--medium--fast converge to desired value
                gain: 1.0
                is_slave: true
        </rosparam>
        <remap from="~imu_filtered" to="~imu" />
    </node> 
</launch>
